tests/jsx/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -1,5 +1,84 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
+exports[`conditional-expression.js 1`] = `
+<div>
+  {isCool ? "cool" : "very nice"}
+</div>;
+
+// should get formatted JSX style
+const a = something ? "string" : <div />;
+
+// should not get formatted JSX style
+const a = something ? "string" : "other string";
+
+<div>
+  {reallyLongChainOfConditionalExpressions ? (
+    "why"
+  ) : yes ? (
+    "it"
+  ) : is ? (
+    "!"
+  ) : (
+    ":)"
+  )}
+</div>;
+
+<div>
+  {reallyLongChainOfConditionalExpressions ? (
+    <span>why</span>
+  ) : yes ? (
+    <span>it</span>
+  ) : is ? (
+    <span>!</span>
+  ) : (
+    <span>:)</span>
+  )}
+</div>;
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+<div>
+  {isCool ? (
```

Comment:
In the case of a null literal as the alternate:
```jsx
{showTheThing ? (
  <Thing />
) : null}
```
I've written that by hand before. I prefer this where possible:
```jsx
{showTheThing && (
  <Thing />
)}
```
But using a ternary with a hanging null protects against creating a child with the value `undefined` (which is disallowed in React). So I can only use `&&` if I know that `showTheThing` will be false or null when not present (and never undefined).

In the case of the string literals, I've voluntarily written that syntax as a human :wink: 
But I think it would be better to make it so that the ternary only breaks and adds parens if it needs to (and one-lines it without parens otherwise). Right now we're *always* breaking and adding parens, but I want to change it to put a group around outermost ConditionalExpression (in a chain of ConditionalExpressions, possibly), and break that whole group so that it's either all on one line, or all of the consequents/alternates have parens around them.